modules/chat/client/controllers/chat.client.controller.js

```diff
@@ -1,43 +1,55 @@
-'use strict';
+(function () {
+  'use strict';
 
-// Create the 'chat' controller
-angular.module('chat').controller('ChatController', ['$scope', '$location', 'Authentication', 'Socket',
-  function ($scope, $location, Authentication, Socket) {
-    // Create a messages array
-    $scope.messages = [];
+  angular
+    .module('chat')
+    .controller('ChatController', ChatController);
 
-    // If user is not signed in then redirect back home
-    if (!Authentication.user) {
-      $location.path('/');
-    }
+  ChatController.$inject = ['$scope', '$state', 'Authentication', 'Socket'];
 
-    // Make sure the Socket is connected
-    if (!Socket.socket) {
-      Socket.connect();
-    }
+  function ChatController($scope, $state, Authentication, Socket) {
+    var vm = this;
+
+    vm.messages = [];
+    vm.messageText = '';
+    vm.sendMessage = sendMessage;
+
+    init();
+
+    function init() {
+      // If user is not signed in then redirect back home
+      if (!Authentication.user) {
+        $state.go('home');
+      }
 
-    // Add an event listener to the 'chatMessage' event
-    Socket.on('chatMessage', function (message) {
-      $scope.messages.unshift(message);
-    });
+      // Make sure the Socket is connected
+      if (!Socket.socket) {
+        Socket.connect();
+      }
+
+      // Add an event listener to the 'chatMessage' event
+      Socket.on('chatMessage', function (message) {
+        vm.messages.unshift(message);
+      });
+
+      // Remove the event listener when the controller instance is destroyed
+      $scope.$on('$destroy', function () {
```

Comment:
Since this is a styling refactor, I thought it would be good to address this as well; if we're strictly adhering to John Papa's style-guide. Might be worthwhile to start the precedent now, with regards to SOC & moving things out into services that shouldn't be in controllers.

It's ok if you want to leave this here though. I don't feel super strong about it :)